File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int l= flowerbed.size();
        //if n=0
         if (n == 0)
            return true;
        
        for (int i = 0; i < l; i++) {
            if (flowerbed[i] == 0) {
                // Check if the left side is empty or it's the first plot
                bool left_empty = (i == 0) || (flowerbed[i - 1] == 0);
                
                // Check if the right side is empty or it's the last plot
                bool right_empty = (i == l - 1) || (flowerbed[i + 1] == 0);
                
                // If both sides are empty, we can place a flower here
                if (left_empty && right_empty) {
                    flowerbed[i] = 1;  // Place a flower
                    n--;  // Decrease the number of flowers to place
                    
                    // If we have placed all flowers, return true
                    if (n == 0)
                        return true;
                }
            }
        }
        
        // If we couldn't place all flowers, return false
        return false;
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        length= len(flowerbed)
        for i in range(length):
            if flowerbed[i]==0:
                left = i==0 or (flowerbed[i-1]==0)
                right = i== (length-1) or (flowerbed[i+1]==0)

                if(left and right):
                    flowerbed[i]=1
                    n-=1
        return n<=0

        